WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift

```
@@ -207,7 +205,7 @@ private extension PaymentCaptureOrchestrator {
 
             switch result {
             case .success:
-                self?.celebrate()
+                self?.celebrate() // plays a sound, haptic
```

💭 Adding `let self = self` to the guard above would _probably_ be worthwhile here. I'm not sure we want to be doing the other things in this block but not saving the receipt, if the `PaymentCaptureOrchestrator` has gone away anyway... 

I traced that back as best I could see and while the `onCompletion` didn't seem like it would currently do anything too surprising in that case, this does seem like somewhere that strange bugs could creep in.

What do you think? Playing devil's advocate I can see the merits of trying to handle the success even if we can't save the receipt too... but generally I'd fall on the side of failing fast(er.)

---

Opened https://github.com/woocommerce/woocommerce-ios/issues/5426